### PR TITLE
Template style tweaks

### PR DESCRIPTION
**Pull request by thullSL**
These were several small tweaks to the template, didn't make sense to create a branch per issue. 

* Parameterising the ec2 instances ssh key pair
* Adding a description to the template
* Assignment of elastic IP is not attempted if "null" is given as the ip

The IP checking is not ideal but parameter conditions seem to be broke AF which i've detailed in the JIRA issue

### CODE REVIEW

**Comment by dwybourn** [comment]
:shipit:

File: dataCenter/singleEcs.json [hunk 1]
```diff
@@ -1,9 +1,13 @@
 {
     "AWSTemplateFormatVersion": "2010-09-09",
+    "Description" : "A stack for the non-persistant, backend infrastructure used by BrsitechSRM",
```
Comment (csscottlogic) [inline dataCenter/singleEcs.json]: "persistent" and "BristechSRM"

File: dataCenter/singleEcs.json [hunk 2]
```diff
@@ -17,8 +21,16 @@
...
             "Description" : "The access key ID with which speaker-comms talks to it's DB"
+        },
+        "speakerServerKeyPair" : {
+            "Type" : "AWS::EC2::KeyPair::KeyName",
+            "Default" : "developmentStack",
+            "Description" : "The key pair used to gain access to the EC2 instance(s) via ssh, should the need arise."
         }
     },
+    "Conditions" : {
+        "asscociateEIP" : {"Fn::Not"  : [{"Fn::Equals" : [ {"Ref" : "PublicElasticIP"}, "null"]}]}
```
Comment (csscottlogic) [inline dataCenter/singleEcs.json]: "associate"

**Comment by csscottlogic** [comment]
I'd say fix the very minor typos and then :shipit:

**Pull request merged**